tests/countries/test_antigua_and_barbuda.py

```diff
@@ -0,0 +1,186 @@
+        )
+        self.assertNoHoliday(
+            "2005-07-04",
+            "2010-07-05",
+            "2015-07-06",
+            "2020-07-06",
+        )
+        self.assertHolidayName(name, range(1955, 2005))
+
+    def test_carnival_monday(self):
+        name = "Carnival Monday"
+        self.assertHolidayName(
+            name,
+            "2021-08-02",
+            "2022-08-01",
+            "2023-08-07",
+            "2024-08-05",
+            "2025-08-04",
+        )
+        self.assertHolidayName(name, range(1955, 2050))
+
+    def test_carnival_tuesday(self):
+        name = "Carnival Tuesday"
+        self.assertNoHoliday("2000-08-01", "2005-08-02")
+        self.assertHolidayName(
+            name,
+            "2021-08-03",
+            "2022-08-02",
+            "2023-08-08",
+            "2024-08-06",
+            "2025-08-05",
+        )
+        self.assertHolidayName(name, range(2006, 2050))
+
+    def test_independence_day(self):
+        name = "Independence Day"
+        self.assertHolidayName(name, (f"{year}-11-01" for year in range(1955, 2050)))
+        dt = ("2009-11-02", "2015-11-02", "2020-11-02")
+        observed_name = f"{name} (observed)"
+        self.assertHolidayName(observed_name, dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_national_heroes_day(self):
+        self.assertNoHoliday(f"{year}-12-09" for year in range(1955, 2005))
+        name = "National Heroes Day"
+        self.assertHolidayName(
+            name,
+            (f"{year}-12-09" for year in range(2005, 2013)),
+        )
+        self.assertNoHolidayName(name, range(2014, 2050))
+
+    def test_sir_vere_cornwall_bird_snr_day(self):
+        name = "Sir Vere Cornwall Bird (SNR) Day"
+        self.assertNoHolidayName(name, range(1955, 2014))
+        self.assertHolidayName(
+            name,
+            (f"{year}-12-09" for year in range(2014, 2050)),
+        )
+
+    def test_christmas_day(self):
+        name = "Christmas Day"
+        self.assertHolidayName(name, (f"{year}-12-25" for year in range(1955, 2050)))
+        dt = (
+            "2004-12-27",
+            "2005-12-27",
+            "2010-12-27",
+            "2011-12-27",
+            "2016-12-27",
+            "2022-12-27",
+        )
+        observed_name = f"{name} (observed)"
+        self.assertHolidayName(observed_name, dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_boxing_day(self):
+        name = "Boxing Day"
+        self.assertHolidayName(name, (f"{year}-12-26" for year in range(1955, 2050)))
+        dt = (
+            "2004-12-28",
+            "2010-12-28",
+            "2021-12-28",
+        )
+        observed_name = f"{name} (observed)"
+        self.assertHolidayName(observed_name, dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_special_holidays(self):
+        self.assertHolidayName(
+            "Day after the General Election",
+            "2018-03-22",
+            "2023-01-19",
+        )
```

Comment:
It's better to move `test_special_holidays()` after `test_no_holidays()`, and add `test_2025()` here.